### ISSUE TITLE
FrameGO: wildly different performance when assigning the index to a column between `index`, `index.values` and `index.to_series`

### ISSUE THREAD

**Issue by zakstucke** [static-frame/static-frame#788]
**Description**

New to SF, looks great. Got much slower performance assigning the index to a column compared to pandas, debugged and realised it was because I was assigning `f["col"] = index.to_series()`, using `.values` directly is the fastest. Simplest reproduction outlined below. 

Is this a bug? Is it safe to assign using `.values` directly?

Thanks

**Example**

```python
import time
import static_frame as sf
import numpy as np

num_rows = 10000000


def get_f():
    return sf.FrameGO(
        index=np.arange(num_rows),
    )


f1 = get_f()
f2 = get_f()
f3 = get_f()

f1_as_series = f1.index.to_series()

before = time.perf_counter()
f1["a"] = f1_as_series
after = time.perf_counter()
print(f"(Series) Elapsed time: {(after - before) * 100:.4f} ms")

before = time.perf_counter()
f2["a"] = f2.index.values
after = time.perf_counter()
print(f"(Values) Elapsed time: {(after - before) * 100:.4f} ms")

before = time.perf_counter()
f3["a"] = f3.index
after = time.perf_counter()
print(f"(direct) Elapsed time: {(after - before) * 100:.4f} ms")

'''
OUT
(Series) Elapsed time: 2.1809 ms
(Values) Elapsed time: 0.0165 ms
(direct) Elapsed time: 254.7633 ms
'''
```

**Platform**

Run the following function (static-frame >= 0.8.1) and provide the results to define your platform and environment:

```python
<Series: platform>
<Index>
platform           Linux-5.19.0-41-generic-x86_64-with-glibc2.35
sys.version        3.11.3 (main, Apr  5 2023, 14:14:37) [GCC 11.3.0]
static-frame       1.3.2
numpy              1.24.2
pandas             1.5.3
xlsxwriter         <ModuleNotFoundError>
openpyxl           <ModuleNotFoundError>
xarray             2023.4.2
tables             <ModuleNotFoundError>
pyarrow            <ModuleNotFoundError>
msgpack            <ModuleNotFoundError>
msgpack_numpy      <ModuleNotFoundError>
<<U13>             <object>
```

**Comment by flexatone** [comment]
Many thanks for your question. I will dig into this and examine more closely, but briefly: assigning `.values` is always faster as we do not check the label alignment and can just simply add the column, so it is only "safe" if you know your order is correct. When assigning a `Series`, we have to check and potentially align labels. But in this case, I assigning the `Index` should not be so slow. Let me take a look and get back to you.

**Comment by flexatone** [comment]
It appears that we have not yet handled the special case of assigning an `Index` into a `FrameGO`, and instead treat it as an arbitrary iterable of Python objects, which means it takes us a long time to convert it to an array and add it to the `FrameGO` (though that is not necessary at all).

I will work on fix. Thanks again for bringing this to my attention.

**Comment by flexatone** [comment]
The fix is complete and will be part of the 1.4.2 release, probably out tomorrow. 

A modified version of your test above shows that assigning an `Index` or `Index.values` to a `FrameGO` now have essentially the same performance. Assigning a `Series`, as expected, has more overhead due to checking for index alignment. 

I welcome any other questions or feedback you might have. 

```python
>>> def get_f():
...     return sf.FrameGO(
...         index=np.arange(10_000_000),
...     )
... 
>>> f1 = get_f()
>>> f2 = get_f()
>>> f3 = get_f()

>>> s1 = f1.index.to_series()
>>> %time f1['a'] = s1
CPU times: user 17.9 ms, sys: 1.23 ms, total: 19.2 ms
Wall time: 18.3 ms

>>> %time f2['a'] = f2.index.values
CPU times: user 22 µs, sys: 19 µs, total: 41 µs
Wall time: 43.4 µs

>>> %time f3['a'] = f3.index
CPU times: user 21 µs, sys: 16 µs, total: 37 µs
Wall time: 39.3 µs
```

**Comment by zakstucke** [comment]
Awesome! Glad I wasn't going mad :)
